pkg/file/file_instance.go

```diff
@@ -0,0 +1,178 @@
+package file
+
+import (
+	"bytes"
+	"encoding/json"
+	"github.com/moov-io/irs/pkg/config"
+	"github.com/moov-io/irs/pkg/records"
+	"github.com/moov-io/irs/pkg/utils"
+)
+
+// File contains the structures of irs file.
+type fileInstance struct {
+	Transmitter    records.Record   `json:"transmitter"`
+	PaymentPersons []*paymentPerson `json:"payment_persons"`
+	EndTransmitter records.Record   `json:"end_transmitter"`
+}
+
+// Validate performs some checks on the file and returns an error if not Validated
+func (f *fileInstance) Validate() error {
+	err := f.validateRecords()
+	if err != nil {
+		return err
+	}
+
+	err = f.validateSequenceNumber()
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// Parse attempts to initialize a *File object assuming the input is valid raw data.
+func (f *fileInstance) Parse(buf []byte) error {
+	bufSize := len(buf)
+	readPtr := 0
+	if string(buf[readPtr]) != config.TRecordType || bufSize < config.RecordLength {
+		return utils.ErrInvalidAscii
+	}
+
+	if f.Transmitter == nil {
+		f.Transmitter = records.NewTRecord()
+	}
+	err := f.Transmitter.Parse(buf[readPtr : readPtr+config.RecordLength])
```

Comment:
Can we add a length check on this?

Reply:
there is length check

Reply:
I don't see a check for `readPtr+config.RecordLength < len(buf)`